sympy/solvers/ode/tests/test_ode.py

```diff
@@ -3876,8 +3877,6 @@ def test_factorable():
     # Unable to get coverage on this without explicit testing because _desolve
     # already handles Pow before we get there but that should be disabled in
     # future so that factorable gets the raw ODE.
-    eq = f(x).diff(x)-1
-    assert _ode_factorable_match(eq**3, f(x), 1) == {'eqns':[eq], 'x0': 1}
```

Comment:
The comments above should be removed. This should be a test for dsolve with the factorable hint

Reply:
@oscarbenjamin when i write this as test then the `classify_ode(eq)` gives factorable as a hint but  `dsolve(eq, hint='factorable')` gives "does not match hint error". 

when i debugged, i got to know that when the user is providing hint then `_desolve` from deutils.py removes the power of equation in preprocessing.
so how should i solve this?

Reply:
Master branch also shows error for the same

Reply:
I don't understand what you mean but it doesn't seem related to this line of code so it should be a comment on the main page of this PR